Objects/TI4_Helpers/TI4_SystemHelper.ttslua

```diff
@@ -502,6 +488,110 @@ local LEADER = {
     HERO = 'hero'
 }
 
+local _objectByNameCache = {}
+local function _getObjectByName(objectName)
+    assert(type(objectName) == 'string')
+
+    -- Check cache for object GUID
+    if _objectByNameCache[objectName] then
+        -- If GUID is present, see if the object is still on the table
+        local object = getObjectFromGUID(_objectByNameCache[objectName])
+        if object then
+            return object
+        else
+            -- If the object wasn't available, clear the GUID that was previously cached
+            _objectByNameCache[objectName] = nil
+        end
+    end
+
+    for _, object in ipairs(getAllObjects()) do
+        if object.getName() == objectName then
+            _objectByNameCache[objectName] = object.getGUID()
+            return object
+        end
+    end
+
+    print('Unable to locate object: ' .. objectName)
```

Comment:
Since this isn't "error" which includes a file and line number, maybe prefix with "SystemHelper:" so easier to figure out where the error is coming from?  (Not a huge deal as the object name should be a strong signal.)